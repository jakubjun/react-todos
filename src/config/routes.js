import notFoundPathName from './constants';
import BaseLayout from '../component/layout/baseLayout/BaseLayout';
import TodoListPage from '../component/page/todoListPage/TodoListPage';
import NotFoundPage from '../component/page/notFoundPage/NotFoundPage';
import TestPage from '../component/page/testPage/TestPage';

const routes = {
  // path, [layout, component]
  '/': [BaseLayout, TodoListPage],
  '/test': [BaseLayout, TestPage],
  '/todo/:id': [BaseLayout, TodoListPage],
  [notFoundPathName]: [BaseLayout, NotFoundPage],
};

export default routes;
