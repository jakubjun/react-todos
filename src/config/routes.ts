import BaseLayout from '../component/layout/baseLayout/BaseLayout';
import TodoListPage from '../component/page/todoListPage/TodoListPage';
import NotFoundPage from '../component/page/notFoundPage/NotFoundPage';
import AuthPage from '../component/page/authPage/AuthPage';

const routes = [
  ['/', BaseLayout, TodoListPage],
  ['/auth', BaseLayout, AuthPage],
];

export const notFoundRoute = [BaseLayout, NotFoundPage];

export default routes;
