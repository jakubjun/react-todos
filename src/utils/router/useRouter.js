import useLocation from './useLocation';
import configRoutes from '../../config/routes';

export const notFoundPathName = 'notfound';

export default function useRouter(routes = configRoutes) {
  const path = useLocation();

  return routes[path] || routes[notFoundPathName];
}
