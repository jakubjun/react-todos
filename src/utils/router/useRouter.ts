import { FC } from 'react';
import useLocation from './useLocation';

export type Route = [string, FC<any>, FC<any>];

export type NotFoundRoute = [FC<any>, FC<any>];

export default function useRouter(
  routes:Route[],
  notFoundRoute: [FC<any>, FC<any>],
): NotFoundRoute {
  const path = useLocation();

  const matchedRoute = routes.find(([pattern]) => {
    if (typeof pattern === 'string') {
      return pattern === path;
    }

    // TODO add support for regex route patterns
    // if (pattern instanceof RegExp) {
    //   return pattern.test(path);
    // }

    return false;
  });

  if (!matchedRoute) {
    return notFoundRoute;
  }

  return [matchedRoute[1], matchedRoute[2]];
}
