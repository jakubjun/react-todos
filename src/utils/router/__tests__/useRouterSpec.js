import renderer from 'react-test-renderer';
import useRouter from '../useRouter';
import { notFoundPathName } from '../../../config/routes';

const routes = {
  // path, [layout, component]
  '/': ['index_layout', 'index_page'],
  // '/todo/:id': [BaseLayout, TodoListPage],
  [notFoundPathName]: ['not_found_layout', 'not_found_page'],
};
describe('useRouter', () => {
  delete window.location;
  window.location = 'http://hello.world/';
  const [layout, page] = useRouter(routes);
  const component = renderer.create(layout).toJSON();

  it('should route', () => {
    expect(component).toMatchSnapshot();
  });
});
