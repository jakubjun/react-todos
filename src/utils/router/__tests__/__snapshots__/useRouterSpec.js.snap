// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useRouter should route 1`] = `"not_found_layout"`;
